sql-parser/src/main/java/io/crate/sql/Literals.java

```diff
@@ -24,14 +24,141 @@
+                    case 'u':
+                    case 'U':
+                        // handle unicode case
+                        final int charsToConsume = (nextChar == 'u') ? 4 : 8;
+                        if (i + 1 + charsToConsume >= length) {
+                            throw new IllegalArgumentException(ESCAPED_UNICODE_ERROR);
+                        }
+                        endIdx = consumeMatchedChars(input,
+                            i + 2,
+                            charsToConsume,
+                            Literals::isHexDigit);
+                        if (endIdx != i + 2 + charsToConsume) {
+                            throw new IllegalArgumentException(ESCAPED_UNICODE_ERROR);
+                        }
+                        builder.appendCodePoint(Integer.parseInt(input, i + 2, endIdx, 16));
+                        i = endIdx - 1; // skip already consumed chars
+                        break;
+                    case 'x':
+                        // handle hex byte case - up to 2 chars for hex value
+                        endIdx = consumeMatchedChars(input,
+                            i + 2,
+                            2,
+                            Literals::isHexDigit);
+                        if (endIdx > i + 2) {
+                            builder.appendCodePoint(Integer.parseInt(input, i + 2, endIdx, 16));
+                            i = endIdx - 1; // skip already consumed chars
+                        } else {
+                            // hex sequence unmatched - output original char
+                            builder.append(nextChar);
+                            i++;
+                        }
+                        break;
+                    case '0':
+                    case '1':
+                    case '2':
+                    case '3':
+                        // handle octal case - up to 3 chars
+                        endIdx = consumeMatchedChars(input,
+                            i + 2,
+                            2,      // first char is already "consumed"
+                            Literals::isOctalDigit);
+                        builder.appendCodePoint(Integer.parseInt(input, i + 1, endIdx, 8));
+                        i = endIdx - 1; // skip already consumed chars
+                        break;
+                    default:
+                        // non-valid escaped char sequence
+                        builder.append(currentChar);
+                }
+            } else {
+                builder.append(currentChar);
+            }
+        }
+        return builder.toString();
+    }
+
+    private static boolean isOctalDigit(final char ch) {
+        return ch >= '0' && ch <= '7';
+    }
+
+    private static boolean isHexDigit(final char ch) {
+        return (ch >= '0' && ch <= '9') || (ch >= 'A' && ch <= 'F') || (ch >= 'a' && ch <= 'f');
+    }
+
+    /**
+     * Consumes a maximum number of characters of a char sequence matching a given predicate
```

Comment:
Thought I had commented on this already, but I guess I didn't save the comment.

It's good to see some comments, but I think in this case the comments are only repeating what the signature already states.

Much more interesting would be *why* it is okay to skip some characters. Maybe include some examples.
The method name is probably also a bit miss-leading. It is not consuming any characters (consume implies the things are being eaten up --> gone afterwards). But it calculates the endIdx / amount of characters that are part of the escape sequence.